### PR TITLE
Switch to getting logs via block hash instead of block number

### PR DESCRIPTION
**Pull request by stwiname**
# Description
Using the block height to get logs will always return results or empty results rather than an invalid result. Switching to using the block hash guarantees we get the right logs for the block. If an endpoint is out of sync and we're unable to get the logs then retry logic will handle it.

Fixes https://github.com/subquery/subql/issues/1999
## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I have tested locally
- [x] I have performed a self review of my changes
- [x] Updated any relevant documentation
- [x] Linked to any relevant issues
- [x] I have added tests relevant to my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] My code is up to date with the base branch
- [x] I have updated relevant changelogs. [We suggest using chan](https://github.com/geut/chan/tree/main/packages/chan)

### CODE REVIEW

**Review by jiqiang90 — APPROVED** [review]
Need benchmark for this change

File: packages/node/src/ethereum/api.ethereum.ts [hunk 1]
```diff
@@ -265,13 +265,8 @@ export class EthereumApi implements ApiWrapper<EthereumBlockWrapper> {
...
     try {
-      const [block, logs] = await Promise.all([
-        this.getBlockPromise(blockNumber, includeTx),
-        this.client.getLogs({
-          fromBlock: hexValue(blockNumber),
-          toBlock: hexValue(blockNumber),
-        }),
-      ]);
+      const block = await this.getBlockPromise(blockNumber, includeTx);
+      const logs = await this.client.getLogs({ blockHash: block.hash });
```
Comment (ianhe8x) [inline packages/node/src/ethereum/api.ethereum.ts:269]: `infura / alchemy` doesn't seem support this parameter.
also, why remove Promise.all?

Comment (stwiname) [reply packages/node/src/ethereum/api.ethereum.ts:269]: We need to get the block hash to make the request

**Comment by stwiname** [comment]
This seems to fail with Alchemy and older blocks (e.g. `0x8243343df08b9751f5ca0c5f8c9c0460d8a9b6351066fae0acbd4d3e776de8bb`) but works with newer blocks

File: packages/node/src/ethereum/api.ethereum.ts (hunk 1 above)
Comment (ianhe8x) [reply packages/node/src/ethereum/api.ethereum.ts:269]: no problem from me then.

**Review by ianhe8x — APPROVED** [review]

**Pull request merged**